packages/foam-vscode/src/core/markdown-provider.ts

```diff
@@ -52,6 +58,7 @@ export class MarkdownResourceProvider implements ResourceProvider {
       onDidChange: (uri: URI) => void;
       onDidCreate: (uri: URI) => void;
       onDidDelete: (uri: URI) => void;
+      onDidRenameFiles: (e: FileRenameEvent) => any;
```

Comment:
I would use a `onDidMove`, as renaming is a subset of moving, and we need to monitor both (e.g. a file that was moved to another folder might need links pointing to it by path to now point to the new location)